cloudformation/cfncluster.cfn.json

```diff
@@ -2899,6 +2912,443 @@
+                  "Ref": "AWS::Region"
+                },
+                "\n",
+                "  fi\n",
+                "  cookbook_url=https://${s3_prefix}.amazonaws.com/cfncluster-resources-",
+                {
+                  "Ref": "AWS::Region"
+                },
+                "/cookbooks/",
+                {
+                  "Fn::FindInMap": [
+                    "CfnClusterVersions",
+                    "default",
+                    "cookbook"
+                  ]
+                },
+                ".tgz\n",
+                "fi\n",
+                "export PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/opt/aws/bin\n",
+                "export cfncluster_version=",
+                {
+                  "Fn::FindInMap": [
+                    "CfnClusterVersions",
+                    "default",
+                    "cfncluster"
+                  ]
+                },
+                "\n",
+                "export cookbook_version=",
+                {
+                  "Fn::FindInMap": [
+                    "CfnClusterVersions",
+                    "default",
+                    "cookbook"
+                  ]
+                },
+                "\n",
+                "export chef_version=",
+                {
+                  "Fn::FindInMap": [
+                    "CfnClusterVersions",
+                    "default",
+                    "chef"
+                  ]
+                },
+                "\n",
+                "export ridley_version=",
+                {
+                  "Fn::FindInMap": [
+                    "CfnClusterVersions",
+                    "default",
+                    "ridley"
+                  ]
+                },
+                "\n",
+                "export berkshelf_version=",
+                {
+                  "Fn::FindInMap": [
+                    "CfnClusterVersions",
+                    "default",
+                    "berkshelf"
+                  ]
+                },
+                "\n",
+                "if [ -f /opt/cfncluster/.bootstrapped ]; then\n",
+                "  installed_version=$(cat /opt/cfncluster/.bootstrapped)\n",
+                "  if [ \"${cfncluster_version}\" != \"${installed_version}\" ]; then\n",
+                "    bootstrap_instance\n",
+                "  fi\n",
+                "else\n",
+                "  bootstrap_instance\n",
+                "fi\n",
+                "mkdir /tmp/cookbooks\n",
+                "cd /tmp/cookbooks\n",
+                "curl -v -L -o /etc/chef/cfncluster-cookbook.tgz -z \"$(cat /etc/chef/cfncluster-cookbook.tgz.date)\" ${cookbook_url}\n",
+                "tar -xzf /etc/chef/cfncluster-cookbook.tgz\n",
+                "cd /tmp\n",
+                "# Call CloudFormation\n",
+                "cfn-init ${proxy_args} -s ",
+                {
+                  "Ref": "AWS::StackName"
+                },
+                " -v -c default -r ",
+                {
```

Comment:
why are we using this condition since we are in the LaunchConfiguration part, can we just leave "ComputeServerLaunchConfig" as below in the LaunchTemplate at line 3787 we just put `ComputeServerLaunchTemplate`

Reply:
fixed.